python/log.py

```diff
@@ -27,6 +30,21 @@
 _output_to_log = False
 
 
+class LogListener(object):
+	def __init__(self, log, close, get_log_level):
+		self._listener = core.BNLogListener()
+		self._listener.context = 0
+		self._listener.log = self._listener.log.__class__(log)
```

Comment:
This is in no way official code review, just my opinion. I think these should be required methods on the LogListener class, rather than parameters to __init__. It'd just be a bit more idiomatic to subclass LogListener and implement some methods rather than instantiating a LogListener with functions as parameters, IMO.

Reply:
That makes sense. What do you mean by "required methods"? Do you mean stub the methods in the LogListener class and have them raise an exception or return NotImplemented or pass or something else?

Reply:
Thanks for the suggestion. It results in a cleaner solution.

Reply:
Should the `LogListener` base class include and reference private methods when registering a log listener in the core lib? For instance, should it do something similar to:
```python
class LogListener(object):
        """Base class for custom LogListener classes"""

        def __init__(self):
                ...
                self._listener.log = self._listener.log.__class__(self._log)

        ...
        def _log(self, ctxt, log_level, msg):
                try:
                        self.log(ctxt, log_level, msg)
                except:
                        log_error(traceback.format_exc())
```

My current implementation doesn't do that. Notice that the `except` block in `_log` calls `log_error(...)`. That is a common design pattern throughout the API, but is, I believe, problematic in the `LogListener` class proposed in this PR when the user's class handles `LogLevel.ErrorLog`.

Is there API to write a message to stderr that doesn't depend on the logging sub-system (i.e., it doesn't reference log levels, etc.)? In this situation, should the Python class avoid going through the core lib and instead just write to stderr?